src/core/gui/dialog/toolbarCustomize/ToolbarAdapter.cpp

```diff
@@ -84,8 +84,10 @@ void ToolbarAdapter::cleanToolItem(GtkToolItem* it) {
     if (data) {
         gtk_widget_set_sensitive(GTK_WIDGET(it), ToolitemDragDrop::isToolItemEnabled(data));
     }
+    GdkWindow* window = gtk_widget_get_window(GTK_WIDGET(it));
 
-    gdk_window_set_cursor(gtk_widget_get_window(GTK_WIDGET(it)), nullptr);
+    if (window)
+        gdk_window_set_cursor(window, nullptr);
```

Comment:
I receive `Gdk-CRITICAL **: 12:18:13.853: gdk_window_set_cursor: assertion 'GDK_IS_WINDOW (window)' failed` for the five GtkToolItem where have a submenu attached. Pen, Eraser, PdfTool, Draw, and Select. This hack remove this - but I don't like the solution. But for now the application can close without giving a SEGV.

Reply:
So should we merge or are you still looking for a better solution? I can confirm that the segfault is fixed.